### ISSUE TITLE
Feature request: Verify URL return codes before shortening on a API key basis

### ISSUE THREAD

**Issue by ScrumpyJack** [shlinkio/shlink#812]
I'd like API key 1 to be able to shorten URL without the URL being checked and API key 2 must have the checked URLs before shortening (returns a 200). 
Hope that makes sense :)

**Comment by acelaya** [comment]
I think that would mix different and unrelated responsibilities.

The purpose of API keys is to handle access control and implicit permissions.

Enabling/disabling features on a per API key basis, would introduce side effects hard to debug.

What can be done (and I have considered it many times) is allowing to pass a flag argument while creating the short URL, which can be used to enable/disable the validation on the fly.

That would make more sense, and would be aligned with many other features in shlink that have a default value at configuration level, but can be overridden at runtime, while creating the URL.

**Comment by acelaya** [comment]
@ScrumpyJack let me know if my proposal works for you, or I'm maybe missing some use case for your request.

**Comment by Tymek** [comment]
> which can be used to enable/disable the URL verification on the fly

good idea. maybe with fallback page/url?

**Comment by acelaya** [comment]
Having a fallback URL might defeat the purpose of having validation.

If long URL you provide is not valid but the short URL is still created, you may not notice it was actually wrong. It's better to not fail silently.

**Comment by ScrumpyJack** [comment]
@acelaya Thanks for taking the time to think about it. 

On my installation of shlink, URL validation is expensive, so I turn it off, but some people want it enforced (ie validation must occur and can not be circumvented, and I can't do that for them on the same instance.

**Comment by acelaya** [comment]
Then my proposal would solve the problem.

You set at config level what you want to usually happen, and teach users to change the value of the option if they want a different behavior (if you are using shlink-web-client, this would also need to be implemented).

Basically, it's not the API who has to handle this through API keys. It would be the wrong solution to the problem, and I can see people opening lots of tickets because of inconsistent behaviors, haha.

Long story short. If you want it to behave in one way or another but transparently, that should in any case be something handled by the front-end consuming shlink, and not by shlink itself.

**Comment by acelaya** [comment]
> Refs #820
